sql: TestTruncateWhileColumnBackfill failed

[(sql).TestTruncateWhileColumnBackfill failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2197065&tab=buildLog) on [master@281285d779fccd0ff89aafda1f1adc261c470d5b](https://github.com/cockroachdb/cockroach/commits/281285d779fccd0ff89aafda1f1adc261c470d5b):

Fatal error:
```
panic: runtime error: index out of range [0] with length 0

```

Stack:
```
goroutine 572437 [running]:
github.com/cockroachdb/cockroach/pkg/sql/sqlbase.(*ImmutableTableDescriptor).MakeFirstMutationPublic(0xc000e64a00, 0x0, 0x0, 0xc00a3be5e0, 0x162cb3d3db910102)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sqlbase/structured.go:3513 +0x424
github.com/cockroachdb/cockroach/pkg/sql.(*SchemaChanger).validateConstraints.func2(0x59ee0a0, 0xc00c504f40, 0xc001a03b00, 0x59ee160)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/backfill.go:569 +0x51
github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1(0xc000ceff68, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:166 +0x3a
golang.org/x/sync/errgroup.(*Group).Go.func1(0xc006cb6ba0, 0xc006cb6c00)
	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57 +0x64
created by golang.org/x/sync/errgroup.(*Group).Go
	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:54 +0x66

```

<details><summary>Log preceding fatal error</summary><p>

```
=== RUN   TestTruncateWhileColumnBackfill

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestTruncateWhileColumnBackfill PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestTruncateWhileColumnBackfill.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql).TestTruncateWhileColumnBackfill failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2197202&tab=buildLog) on [master@f39153f28a28d7bef8bcaeaaa7e41a2aec218aa9](https://github.com/cockroachdb/cockroach/commits/f39153f28a28d7bef8bcaeaaa7e41a2aec218aa9):

Fatal error:
```
panic: runtime error: index out of range [0] with length 0

```

Stack:
```
goroutine 578823 [running]:
github.com/cockroachdb/cockroach/pkg/sql/sqlbase.(*ImmutableTableDescriptor).MakeFirstMutationPublic(0xc00571b400, 0x0, 0x0, 0xc0080fbda0, 0x162cb4fe34ef14ce)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sqlbase/structured.go:3513 +0x424
github.com/cockroachdb/cockroach/pkg/sql.(*SchemaChanger).validateConstraints.func2(0x59ee2e0, 0xc001feb9c0, 0xc005cac380, 0x59ee3a0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/backfill.go:569 +0x51
github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1(0xc00009bf68, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:166 +0x3a
golang.org/x/sync/errgroup.(*Group).Go.func1(0xc00a5c55c0, 0xc00a5c5620)
	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57 +0x64
created by golang.org/x/sync/errgroup.(*Group).Go
	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:54 +0x66

```

<details><summary>Log preceding fatal error</summary><p>

```
=== RUN   TestTruncateWhileColumnBackfill

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestTruncateWhileColumnBackfill PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestTruncateWhileColumnBackfill.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql).TestTruncateWhileColumnBackfill failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2197415&tab=buildLog) on [master@697e455f453581cf450d0bbf2903a00c6d9ebffd](https://github.com/cockroachdb/cockroach/commits/697e455f453581cf450d0bbf2903a00c6d9ebffd):

```
=== RUN   TestTruncateWhileColumnBackfill
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTruncateWhileColumnBackfill458731528
--- FAIL: TestTruncateWhileColumnBackfill (1.09s)
    test_log_scope.go:154: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTruncateWhileColumnBackfill458731528
    test_log_scope.go:63: use -show-logs to present logs inline
    schema_changer_test.go:4278: expected 0 key value pairs, but got 4001


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestTruncateWhileColumnBackfill PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestTruncateWhileColumnBackfill.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql).TestTruncateWhileColumnBackfill failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2197678&tab=buildLog) on [master@935ae2e97d5c3ea61b93a8639f8d695fbc137bf8](https://github.com/cockroachdb/cockroach/commits/935ae2e97d5c3ea61b93a8639f8d695fbc137bf8):

```
=== RUN   TestTruncateWhileColumnBackfill
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTruncateWhileColumnBackfill833954438
--- FAIL: TestTruncateWhileColumnBackfill (1.09s)
    test_log_scope.go:154: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTruncateWhileColumnBackfill833954438
    test_log_scope.go:63: use -show-logs to present logs inline
    schema_changer_test.go:4278: expected 0 key value pairs, but got 4001


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestTruncateWhileColumnBackfill PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestTruncateWhileColumnBackfill.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql).TestTruncateWhileColumnBackfill failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2197055&tab=buildLog) on [master@281285d779fccd0ff89aafda1f1adc261c470d5b](https://github.com/cockroachdb/cockroach/commits/281285d779fccd0ff89aafda1f1adc261c470d5b):

```
=== RUN   TestTruncateWhileColumnBackfill
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTruncateWhileColumnBackfill622062862
--- FAIL: TestTruncateWhileColumnBackfill (5.49s)
    test_log_scope.go:154: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTruncateWhileColumnBackfill622062862
    test_log_scope.go:63: use -show-logs to present logs inline
    schema_changer_test.go:4278: expected 0 key value pairs, but got 301


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestTruncateWhileColumnBackfill PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestTruncateWhileColumnBackfill.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql).TestTruncateWhileColumnBackfill failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2197753&tab=buildLog) on [master@51b0af472fa738d0318af7860874c456f8838f83](https://github.com/cockroachdb/cockroach/commits/51b0af472fa738d0318af7860874c456f8838f83):

Fatal error:
```
panic: runtime error: index out of range [0] with length 0

```

Stack:
```
goroutine 583423 [running]:
github.com/cockroachdb/cockroach/pkg/sql/sqlbase.(*ImmutableTableDescriptor).MakeFirstMutationPublic(0xc00238c500, 0x0, 0x0, 0xc00aafcce0, 0x162cb9c8f7f376c9)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sqlbase/structured.go:3513 +0x424
github.com/cockroachdb/cockroach/pkg/sql.(*SchemaChanger).validateConstraints.func2(0x59f22c0, 0xc0057ca380, 0xc003914380, 0x59f2380)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/backfill.go:569 +0x51
github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1(0xc000cb8768, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:166 +0x3a
golang.org/x/sync/errgroup.(*Group).Go.func1(0xc005748000, 0xc005748060)
	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57 +0x64
created by golang.org/x/sync/errgroup.(*Group).Go
	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:54 +0x66

```

<details><summary>Log preceding fatal error</summary><p>

```
=== RUN   TestTruncateWhileColumnBackfill

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestTruncateWhileColumnBackfill PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestTruncateWhileColumnBackfill.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql).TestTruncateWhileColumnBackfill failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2197799&tab=buildLog) on [master@697e455f453581cf450d0bbf2903a00c6d9ebffd](https://github.com/cockroachdb/cockroach/commits/697e455f453581cf450d0bbf2903a00c6d9ebffd):

```
=== RUN   TestTruncateWhileColumnBackfill
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTruncateWhileColumnBackfill351843665
--- FAIL: TestTruncateWhileColumnBackfill (1.19s)
    test_log_scope.go:154: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTruncateWhileColumnBackfill351843665
    test_log_scope.go:63: use -show-logs to present logs inline
    schema_changer_test.go:4278: expected 0 key value pairs, but got 4001


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestTruncateWhileColumnBackfill PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestTruncateWhileColumnBackfill.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql).TestTruncateWhileColumnBackfill failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2197193&tab=buildLog) on [master@f39153f28a28d7bef8bcaeaaa7e41a2aec218aa9](https://github.com/cockroachdb/cockroach/commits/f39153f28a28d7bef8bcaeaaa7e41a2aec218aa9):

```
=== RUN   TestTruncateWhileColumnBackfill
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTruncateWhileColumnBackfill686144483
--- FAIL: TestTruncateWhileColumnBackfill (5.29s)
    test_log_scope.go:154: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTruncateWhileColumnBackfill686144483
    test_log_scope.go:63: use -show-logs to present logs inline
    schema_changer_test.go:4278: expected 0 key value pairs, but got 301


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestTruncateWhileColumnBackfill PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestTruncateWhileColumnBackfill.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql).TestTruncateWhileColumnBackfill failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2197819&tab=buildLog) on [master@c5fd3de804c0f39e550505ccaf84be3197cb99e3](https://github.com/cockroachdb/cockroach/commits/c5fd3de804c0f39e550505ccaf84be3197cb99e3):

```
=== RUN   TestTruncateWhileColumnBackfill
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTruncateWhileColumnBackfill555753503
--- FAIL: TestTruncateWhileColumnBackfill (2.06s)
    test_log_scope.go:154: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTruncateWhileColumnBackfill555753503
    test_log_scope.go:63: use -show-logs to present logs inline
    schema_changer_test.go:4278: expected 0 key value pairs, but got 4001


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestTruncateWhileColumnBackfill PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestTruncateWhileColumnBackfill.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql).TestTruncateWhileColumnBackfill failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2197867&tab=buildLog) on [master@e2f93484ccbb35fdb5f218b36255f1da9e4d2867](https://github.com/cockroachdb/cockroach/commits/e2f93484ccbb35fdb5f218b36255f1da9e4d2867):

Fatal error:
```
panic: runtime error: index out of range [0] with length 0

```

Stack:
```
goroutine 620232 [running]:
github.com/cockroachdb/cockroach/pkg/sql/sqlbase.(*ImmutableTableDescriptor).MakeFirstMutationPublic(0xc004a9c500, 0x0, 0x0, 0xc00a4de080, 0x162cbbb75b6bc660)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sqlbase/structured.go:3513 +0x424
github.com/cockroachdb/cockroach/pkg/sql.(*SchemaChanger).validateConstraints.func2(0x59f4220, 0xc007006640, 0xc0048d9600, 0x59f42e0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/backfill.go:569 +0x51
github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1(0xc0004b1f68, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:166 +0x3a
golang.org/x/sync/errgroup.(*Group).Go.func1(0xc00765f890, 0xc00765f8f0)
	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57 +0x64
created by golang.org/x/sync/errgroup.(*Group).Go
	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:54 +0x66

```

<details><summary>Log preceding fatal error</summary><p>

```
=== RUN   TestTruncateWhileColumnBackfill

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestTruncateWhileColumnBackfill PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestTruncateWhileColumnBackfill.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql).TestTruncateWhileColumnBackfill failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2198009&tab=buildLog) on [master@d89d18bd2db88034463f2d1379190e6915309939](https://github.com/cockroachdb/cockroach/commits/d89d18bd2db88034463f2d1379190e6915309939):

Fatal error:
```
panic: runtime error: index out of range [0] with length 0

```

Stack:
```
goroutine 582154 [running]:
github.com/cockroachdb/cockroach/pkg/sql/sqlbase.(*ImmutableTableDescriptor).MakeFirstMutationPublic(0xc001c03900, 0x0, 0x0, 0xc00d312420, 0x162cbd2901bc3512)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sqlbase/structured.go:3513 +0x424
github.com/cockroachdb/cockroach/pkg/sql.(*SchemaChanger).validateConstraints.func2(0x59f4000, 0xc00851ff80, 0xc003876d80, 0x59f40c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/backfill.go:569 +0x51
github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1(0xc000c9d768, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:166 +0x3a
golang.org/x/sync/errgroup.(*Group).Go.func1(0xc00392d8c0, 0xc00392d920)
	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57 +0x64
created by golang.org/x/sync/errgroup.(*Group).Go
	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:54 +0x66

```

<details><summary>Log preceding fatal error</summary><p>

```
=== RUN   TestTruncateWhileColumnBackfill

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestTruncateWhileColumnBackfill PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestTruncateWhileColumnBackfill.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql).TestTruncateWhileColumnBackfill failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2197668&tab=buildLog) on [master@935ae2e97d5c3ea61b93a8639f8d695fbc137bf8](https://github.com/cockroachdb/cockroach/commits/935ae2e97d5c3ea61b93a8639f8d695fbc137bf8):

```
=== RUN   TestTruncateWhileColumnBackfill
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTruncateWhileColumnBackfill759835125
--- FAIL: TestTruncateWhileColumnBackfill (7.56s)
    test_log_scope.go:154: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTruncateWhileColumnBackfill759835125
    test_log_scope.go:63: use -show-logs to present logs inline
    schema_changer_test.go:4278: expected 0 key value pairs, but got 301


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestTruncateWhileColumnBackfill PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestTruncateWhileColumnBackfill.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql).TestTruncateWhileColumnBackfill failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2197743&tab=buildLog) on [master@51b0af472fa738d0318af7860874c456f8838f83](https://github.com/cockroachdb/cockroach/commits/51b0af472fa738d0318af7860874c456f8838f83):

```
=== RUN   TestTruncateWhileColumnBackfill
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTruncateWhileColumnBackfill271607236
--- FAIL: TestTruncateWhileColumnBackfill (6.51s)
    test_log_scope.go:154: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTruncateWhileColumnBackfill271607236
    test_log_scope.go:63: use -show-logs to present logs inline
    schema_changer_test.go:4278: expected 0 key value pairs, but got 301


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestTruncateWhileColumnBackfill PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestTruncateWhileColumnBackfill.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql).TestTruncateWhileColumnBackfill failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2197809&tab=buildLog) on [master@c5fd3de804c0f39e550505ccaf84be3197cb99e3](https://github.com/cockroachdb/cockroach/commits/c5fd3de804c0f39e550505ccaf84be3197cb99e3):

Fatal error:
```
panic: runtime error: index out of range [0] with length 0

```

Stack:
```
goroutine 618382 [running]:
github.com/cockroachdb/cockroach/pkg/sql/sqlbase.(*ImmutableTableDescriptor).MakeFirstMutationPublic(0xc02cd28f00, 0x0, 0x0, 0xc00bd72620, 0x162cbf170385748a)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sqlbase/structured.go:3513 +0x6f5
github.com/cockroachdb/cockroach/pkg/sql.(*SchemaChanger).validateConstraints.func2(0x8fd5800, 0xc02cd2a600, 0xc02cf84d80, 0xc11aba)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/backfill.go:569 +0x96
github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1(0xc00c4ff768, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:166 +0x4f
golang.org/x/sync/errgroup.(*Group).Go.func1(0xc02d26a1b0, 0xc02d26a210)
	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57 +0x72
created by golang.org/x/sync/errgroup.(*Group).Go
	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:54 +0x74

```

<details><summary>Log preceding fatal error</summary><p>

```
=== RUN   TestTruncateWhileColumnBackfill

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestTruncateWhileColumnBackfill PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestTruncateWhileColumnBackfill.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql).TestTruncateWhileColumnBackfill failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2198284&tab=buildLog) on [master@f89dbd928242862863858483f6970854a7cc912a](https://github.com/cockroachdb/cockroach/commits/f89dbd928242862863858483f6970854a7cc912a):

```
=== RUN   TestTruncateWhileColumnBackfill
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTruncateWhileColumnBackfill129990090
--- FAIL: TestTruncateWhileColumnBackfill (3.21s)
    test_log_scope.go:154: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTruncateWhileColumnBackfill129990090
    test_log_scope.go:63: use -show-logs to present logs inline
    schema_changer_test.go:4278: expected 0 key value pairs, but got 4001


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestTruncateWhileColumnBackfill PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestTruncateWhileColumnBackfill.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql).TestTruncateWhileColumnBackfill failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2197858&tab=buildLog) on [master@e2f93484ccbb35fdb5f218b36255f1da9e4d2867](https://github.com/cockroachdb/cockroach/commits/e2f93484ccbb35fdb5f218b36255f1da9e4d2867):

```
=== RUN   TestTruncateWhileColumnBackfill
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTruncateWhileColumnBackfill901602802
--- FAIL: TestTruncateWhileColumnBackfill (8.69s)
    test_log_scope.go:154: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTruncateWhileColumnBackfill901602802
    test_log_scope.go:63: use -show-logs to present logs inline
    schema_changer_test.go:4278: expected 0 key value pairs, but got 301


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestTruncateWhileColumnBackfill PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestTruncateWhileColumnBackfill.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/sql/schema_changer_test.go
+++ b/pkg/sql/schema_changer_test.go
@@ -668,11 +668,6 @@ CREATE UNIQUE INDEX vidx ON t.test (v);
 func TestDropWhileBackfill(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	defer func(oldInterval time.Duration) {
-		jobs.DefaultAdoptInterval = oldInterval
-	}(jobs.DefaultAdoptInterval)
-	jobs.DefaultAdoptInterval = 50 * time.Millisecond
-
 	// protects backfillNotification
 	var mu syncutil.Mutex
 	backfillNotification := make(chan struct{})
@@ -4192,28 +4187,28 @@ func TestTruncateWhileColumnBackfill(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	defer func(oldInterval time.Duration) {
-		jobs.DefaultAdoptInterval = oldInterval
-	}(jobs.DefaultAdoptInterval)
-	jobs.DefaultAdoptInterval = 50 * time.Millisecond
+	skip.WithIssue(t, 43990)
 
 	backfillNotification := make(chan struct{})
 	backfillCount := int64(0)
 	params, _ := tests.CreateTestServerParams()
 	params.Knobs = base.TestingKnobs{
+		// Runs schema changes asynchronously.
 		SQLSchemaChanger: &sql.SchemaChangerTestingKnobs{
-			BackfillChunkSize: 100,
+			// TODO (lucy): if/when this test gets reinstated, figure out what knobs are
+			// needed.
 		},
 		DistSQL: &execinfra.TestingKnobs{
 			RunBeforeBackfillChunk: func(sp roachpb.Span) error {
 				switch atomic.LoadInt64(&backfillCount) {
-				case 2:
+				case 3:
 					// Notify in the middle of a backfill.
 					if backfillNotification != nil {
 						close(backfillNotification)
 						backfillNotification = nil
 					}
-					return nil
+					// Never complete the backfill.
+					return context.DeadlineExceeded
 				default:
 					atomic.AddInt64(&backfillCount, 1)
 				}
@@ -4232,44 +4227,44 @@ CREATE TABLE t.test (k INT PRIMARY KEY, v INT);
 	}
 
 	// Bulk insert.
-	maxValue := 4000
-	if util.RaceEnabled {
-		// Race builds are a lot slower, so use a smaller number of rows.
-		// We expect this to also reduce the memory footprint of the test.
-		maxValue = 300
-	}
+	const maxValue = 5000
 	if err := bulkInsertIntoTable(sqlDB, maxValue); err != nil {
 		t.Fatal(err)
 	}
 
 	notify := backfillNotification
 
-	var wg sync.WaitGroup
-	wg.Add(1)
-	const add_column = `ALTER TABLE t.test ADD COLUMN x DECIMAL NOT NULL DEFAULT 1.4::DECIMAL, ADD CHECK (x >= 0)`
-	var addColErr error
-	go func() {
-		_, addColErr = sqlDB.Exec(add_column)
-		wg.Done()
-	}()
+	const add_column = `ALTER TABLE t.public.test ADD COLUMN x DECIMAL NOT NULL DEFAULT 1.4::DECIMAL, ADD CHECK (x >= 0)`
+	if _, err := sqlDB.Exec(add_column); err != nil {
+		t.Fatal(err)
+	}
 
-	const drop_column = `ALTER TABLE t.test DROP COLUMN v`
+	const drop_column = `ALTER TABLE t.public.test DROP COLUMN v`
 	if _, err := sqlDB.Exec(drop_column); err != nil {
 		t.Fatal(err)
 	}
 
-	// Wait for backfill to have started.
-	<-notify
-	if _, err := sqlDB.Exec("TRUNCATE TABLE t.test"); err != nil {
-		t.Fatal(err)
+	// Check that an outstanding schema change exists.
+	tableDesc := catalogkv.TestingGetTableDescriptor(kvDB, keys.SystemSQLCodec, "t", "test")
+	oldID := tableDesc.ID
+	if lenMutations := len(tableDesc.Mutations); lenMutations != 3 {
+		t.Fatalf("%d outstanding schema change", lenMutations)
 	}
 
-	// Wait for ADD COLUMN to complete.
+	// Run TRUNCATE.
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		<-notify
+		if _, err := sqlDB.Exec("TRUNCATE TABLE t.test"); err != nil {
+			t.Error(err)
+		}
+		wg.Done()
+	}()
 	wg.Wait()
-	require.NoError(t, addColErr)
 
 	// The new table is truncated.
-	tableDesc := catalogkv.TestingGetTableDescriptor(kvDB, keys.SystemSQLCodec, "t", "test")
+	tableDesc = catalogkv.TestingGetTableDescriptor(kvDB, keys.SystemSQLCodec, "t", "test")
 	tablePrefix := keys.SystemSQLCodec.TablePrefix(uint32(tableDesc.ID))
 	tableEnd := tablePrefix.PrefixEnd()
 	if kvs, err := kvDB.Scan(context.Background(), tablePrefix, tableEnd, 0); err != nil {
@@ -4290,6 +4285,26 @@ CREATE TABLE t.test (k INT PRIMARY KEY, v INT);
 	}
 	if checks := tableDesc.AllActiveAndInactiveChecks(); len(checks) != 1 {
 		t.Fatalf("expected 1 check, found %d", len(checks))
+	}
+
+	sqlRun := sqlutils.MakeSQLRunner(sqlDB)
+	if err := jobutils.VerifySystemJob(t, sqlRun, 0, jobspb.TypeSchemaChange, jobs.StatusSucceeded, jobs.Record{
+		Username:    security.RootUser,
+		Description: add_column,
+		DescriptorIDs: descpb.IDs{
+			oldID,
+		},
+	}); err != nil {
+		t.Fatal(err)
+	}
+	if err := jobutils.VerifySystemJob(t, sqlRun, 1, jobspb.TypeSchemaChange, jobs.StatusSucceeded, jobs.Record{
+		Username:    security.RootUser,
+		Description: drop_column,
+		DescriptorIDs: descpb.IDs{
+			oldID,
+		},
+	}); err != nil {
+		t.Fatal(err)
 	}
 }
 

--- a/pkg/sql/sqlbase/structured.go
+++ b/pkg/sql/sqlbase/structured.go
@@ -3056,24 +3056,21 @@ func (desc *MutableTableDescriptor) MakeMutationComplete(m descpb.DescriptorMuta
 		case *descpb.DescriptorMutation_Constraint:
 			switch t.Constraint.ConstraintType {
 			case descpb.ConstraintToUpdate_CHECK:
-				if t.Constraint.Check.Validity == descpb.ConstraintValidity_Dropping {
-					break
-				}
-				var found bool
-				for _, c := range desc.Checks {
-					if c.Name == t.Constraint.Name {
-						if t.Constraint.Check.Validity != descpb.ConstraintValidity_Unvalidated {
+				switch t.Constraint.Check.Validity {
+				case descpb.ConstraintValidity_Validating:
+					// Constraint already added, just mark it as Validated
+					for _, c := range desc.Checks {
+						if c.Name == t.Constraint.Name {
 							c.Validity = descpb.ConstraintValidity_Validated
+							break
 						}
-						found = true
-						break
 					}
-				}
-				if !found {
-					if t.Constraint.Check.Validity != descpb.ConstraintValidity_Unvalidated {
-						t.Constraint.Check.Validity = descpb.ConstraintValidity_Validated
-					}
+				case descpb.ConstraintValidity_Unvalidated:
+					// add the constraint to the list of check constraints on the table
+					// descriptor
 					desc.Checks = append(desc.Checks, &t.Constraint.Check)
+				default:
+					return errors.AssertionFailedf("invalid constraint validity state: %d", t.Constraint.Check.Validity)
 				}
 			case descpb.ConstraintToUpdate_FOREIGN_KEY:
 				switch t.Constraint.ForeignKey.Validity {

